entrofy/mappers.py

```diff
@@ -0,0 +1,242 @@
+
+        return df
+
+
+class ObjectMapper(BaseMapper):
+    '''A generic object-mapper.
+
+    This mapper is appropriate for strings or categorical types.
+
+    Attributes
+    ----------
+    targets : dict
+        A dictionary mapping output column names to target probabilities
+    '''
+    def __init__(self, column, prefix='', n_out=None, targets=None):
+        '''Object mapper.
+
+        Parameters
+        ----------
+        column : pd.Series
+            A pandas Series object
+
+        prefix : str
+            A string to prepend to output column names
+
+        n_out : int or None
+            maximum number of output columns to generate.
+            If fewer than the number of unique values in `column`, the
+            most populous values are selected first.
+
+        targets: dict {value: probability}
+            An optional pre-computed target dictionary.
+            If provided, then `n_out` and `prefix` are ignored.
+        '''
+        if targets is not None:
+            self.targets = targets
+            self._map = {v: equal_maker(v) for v in targets}
+        else:
+            # 1. determine unique values
+            values = column.value_counts().index
+
+            if n_out is None:
+                n_out = len(values)
+
+            # 2. build targets dict
+            self.targets = {}
+            self._map = {}
+
+            values = values[:n_out]
+            target_prob = 1./len(values)
+
+            for val in values:
+                key = '{}{}'.format(prefix, val)
+                self.targets[key] = target_prob
+                self._map[key] = equal_maker(val)
+
+
+class ContinuousMapper(BaseMapper):
+    """
+    Map continuous values into a set of discrete bins.
+
+    """
+
+    def __init__(self, df, n_out=2, boundaries=None, targets=None,
+                 column_names=None, prefix=""):
+        """
+        This class maps continuous values into a set of `n_out` discrete bins.
+
+        Parameters
+        ----------
+        df : pandas.DataFrame
+            A DataFrame with a single column containing the relevant data
+
+        n_out: int, optional, default: 1
+            The number of discrete bins to use
+
+        boundaries: iterable, optional
+            If `boundaries` is set to a list of lower and upper bin edges,
+            then these edges will be used. Must be of len(boundaries) == n_out+1
+            If `None`, the data will be split up into `n_out` equal-sized bins
+
+        column_names: iterable, optional
+            An optional list of strings with the names for the individual
+            columns created in this class. Must be of length `n_out`. If None,
+            the values of the ranges will be used.
+
+        prefix: string, optional
+            A prefix string to go in front of the `column_names`
+
+        """
+
+        self.n_out = n_out
+        self.prefix = prefix
+
+        minval = np.nanmin(np.array(df))
```

Comment:
I think we need to specify that Mappers operate on series, not entire dataframes.